pkg/server/commands/listusers/list_users_rpc.go

```diff
@@ -59,6 +61,13 @@ func NewListUsersQuery(ds storage.RelationshipTupleReader, opts ...ListUsersQuer
 	return l
 }
 
+// WithResolveNodeLimit see server.WithResolveNodeLimit
+func WithResolveNodeLimit(limit uint32) ListUsersQueryOption {
+	return func(d *listUsersQuery) {
+		d.resolveNodeLimit = limit
+	}
+}
```

Comment:
@miparnisari @jon-whit @sergiught - I repealed the previous change which unnecessarily exported `ListUsersRequest`. Instead, I'm following the pattern set by ListObjects and inject the max depth limit with `WithResolveNodeLimit`.